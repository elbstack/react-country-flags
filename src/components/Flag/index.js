import React, { Component, createElement } from 'react';
import PropTypes from 'prop-types';
import flags from '../../flags/react';

const fetchFlag = async country => {
  if (!flags[country]) return Promise.resolve(null);
  const { default: data } = await flags[country]();
  return data;
};

export const countries = (() => Object.keys(flags))();

export default class Flags extends Component {
  state = {
    flag: null,
  };

  static defaultProps = {
    country: null,
    asSquare: false,
  };

  componentDidMount() {
    this.updateFlag();
  }

  componentDidUpdate(prevProps) {
    const { country } = this.props;

    if (country !== prevProps.country) {
      this.updateFlag();
    }
  }

  async updateFlag() {
    const { country } = this.props;
    const flag = await fetchFlag(country);
    this.setState({ flag });
  }

  render() {
    const { asSquare } = this.props;
    const { flag } = this.state;
    if (!flag) return null;
    if (!asSquare) return createElement(flag);
    return (
      <svg viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
        {createElement(flag)}
      </svg>
    );
  }
}

Flags.propTypes = {
  country: PropTypes.string,
  asSquare: PropTypes.bool,
};
