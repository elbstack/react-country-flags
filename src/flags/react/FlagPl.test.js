import React from 'react';
import renderer from 'react-test-renderer';
import Flag from './FlagPl';

it('FlagPl renders without crashing', () => {
  const component = renderer.create(<Flag />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
