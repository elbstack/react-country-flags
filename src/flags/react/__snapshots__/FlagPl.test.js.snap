// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagPl renders without crashing 1`] = `
<svg
  viewBox="0 0 16 10"
>
  <path
    d="M0 0h16v10H0z"
    fill="#fff"
  />
  <path
    d="M0 5h16v5H0z"
    fill="#dc143c"
  />
</svg>
`;
